README.md

```diff
@@ -269,106 +269,74 @@ The built-in multi-cursor support may not meet your needs. Please refer to the p
-> when running a command before exiting visual mode.
-
-#### Examples
-
-Format selection (default binding):
+Load the module:
 
-```vim
-xnoremap = <Cmd>call VSCodeCall('editor.action.formatSelection')<CR>
-nnoremap = <Cmd>call VSCodeCall('editor.action.formatSelection')<CR><Esc>
-nnoremap == <Cmd>call VSCodeCall('editor.action.formatSelection')<CR>
-```
-
-Open definition aside (default binding):
-
-```vim
-nnoremap <C-w>gd <Cmd>call VSCodeNotify('editor.action.revealDefinitionAside')<CR>
-```
-
-Find in files for word under cursor:
-
-```vim
-nnoremap ? <Cmd>call VSCodeNotify('workbench.action.findInFiles', { 'query': expand('<cword>')})<CR>
+```lua
+local vscode = require('vscode-neovim')
 ```
 
-More advanced examples can be found [here](https://github.com/vscode-neovim/vscode-neovim/tree/master/vim).
-
-### Lua
-
-Load module: `local vscode = require("vscode-neovim")`
-
-1. `vscode.action` for asynchronous execution of actions.
-2. `vscode.call` for synchronous execution of actions.
-3. `vscode.on` for adding hook functions.
-4. `vscode.has_config` check if a configuration exists
-5. `vscode.get_config` get a configuration value
-6. `vscode.update_config` update a configuration
-7. `vscode.notify` like `vim.notify`, but use vscode notification to show the message
-8. `vscode.to_op` A helper for `map-operator`. See [code_actions.lua](./runtime/plugin/code_actions.lua) for the usage
-9. `vscode.get_status_item` Creates a status item
-
-#### Actions
+1. `vscode.action()`: asynchronously executes a vscode command.
+2. `vscode.call()`: synchronously executes a vscode command.
+3. `vscode.on()`: defines a handler for some Nvim UI events.
+4. `vscode.has_config()`: checks if a vscode setting exists.
+5. `vscode.get_config()`: gets a vscode setting value.
+6. `vscode.update_config()`: sets a vscode setting.
+7. `vscode.notify()`: shows a vscode message (see also Nvim's `vim.notify`).
+8. `vscode.to_op()`: A helper for `map-operator`. See [code_actions.lua](./runtime/plugin/code_actions.lua) for the usage
+9. `vscode.get_status_item`: Gets a vscode statusbar item. Properties can be assigned, which magically updates the statusbar item.
 
-##### `vscode.action(name, opts)`
+### vscode.action(name, opts)
 
-This function is used to run an action asynchronously.
+Asynchronously executes a vscode command.
 
 Parameters:
 
 -   `name` (string): The name of the action, generally a vscode command.
--   `opts` (table): Optional table of options. All fields in the table are optional.
-    -   `args` (table): Optional arguments for the action.
-    -   `range` (table): Specific range for the action. In visual mode, this parameter is generally not needed. There
-        are three supported formats for defining the range (all values are 0-indexed):
+-   `opts` (table): Map of optional parameters:
+    -   `args` (table): List of argument `{name = value}` pairs, passed to the action.
```

Comment:
Args only needs to be an array

`args = {1， 2， 3}`
`args = {{name=value}}`

Reply:
Does this mapping work for you?

    nnoremap ? <Cmd>lua require('vscode-neovim').action('workbench.action.findInFiles', { args = { vim.fn.expand('<cword>') } })<CR>

Reply:
You misunderstood the meaning of "query". The parameter received by the findInFiles command is an object, and one of the fields is "query".

Reply:
try
```lua
local arg = {
  query = vim.fn.expand('<cword>'),
  replace = 'abc',
}
require('vscode-neovim').action('workbench.action.findInFiles', {
  args = { arg },
})
```

Reply:
You said:

> Args only needs to be an array `args = {1， 2， 3}`

but the last example shows a list of key=value pairs. Can you show how the mapping would be defined with only a list of values, _not_ a list of key=value pairs?

Reply:
Oh I get it: the keyword arg is expected by the vscode command itself, in this case.

Reply:
Yea